### PR TITLE
Quick Navigation user should be able to tab from list of results to preview and use arrow keys within the list

### PR DESCRIPTION
**Pull request by marinaaisa**
Bug/issue #111742596, if applicable: 

## Summary

Quick Navigation user should be able to tab from list of results to preview and use arrow keys within the list, to avoid tabbing all the list

## Dependencies

NA

## Testing

Steps:
1. Go to any documentation page
2. Open Quick Navigation
3. Assert that you can tab from the list of results to the preview without tabbing all the list
4. Assert that you can move in the list of results using your arrow keys

## Checklist

Make sure you check off the following items. If they cannot be completed, provide a reason.

- [x] Added tests
- [x] Ran `npm test`, and it succeeded
- [x] Updated documentation if necessary

### CODE REVIEW

**Comment by hqhhuang** [comment]
@swift-ci test

**Comment by marinaaisa** [comment]
To do: fix tests

File: src/components/Navigator/QuickNavigationModal.vue [hunk 1]
```diff
@@ -355,11 +356,17 @@ export default {
...
+    },
+    handleDownKeyInput() {
+      this.focusedIndex = 0;
+      this.focusReference();
     },
-    scrollIntoView(index) {
-      this.$refs.match[index].$el.scrollIntoView({
+    scrollIntoView() {
+      if (!this.$refs.match || this.$refs.match[this.focusedIndex]) return false;
+      return this.$refs.match[this.focusedIndex].scrollIntoView({
```
Comment (dobromir-hristov) [inline src/components/Navigator/QuickNavigationModal.vue:369]: Why sometimes `this.$refs.match[this.focusedIndex]` and others `this.$refs.match[this.focusedIndex].$el` ?

I feel like you could create a computed for this 🤔

File: src/components/Navigator/QuickNavigationModal.vue [hunk 2]
```diff
@@ -311,6 +307,11 @@ export default {
     formatSymbolTitle(symbolTitle, symbolStart, symbolEnd) {
       return symbolTitle.slice(symbolStart, symbolEnd);
     },
+    isFocused(index) {
+      if (this.focusedInput || this.focusedIndex !== index) return false;
+      this.focusReference();
```
Comment (dobromir-hristov) [inline src/components/Navigator/QuickNavigationModal.vue:312]: `isFocused` is used in the template, on the `tabindex`.

I know generally it should be fine to do this, but we are relying on the vue template reactivity to call a mutation method. 

Vue caches or hoists templates in various ways and I dont know what could go wrong here.

File: src/components/Navigator/QuickNavigationModal.vue [hunk 3]
```diff
@@ -63,13 +59,17 @@
             <div
               v-bind="{[SCROLL_LOCK_DISABLE_ATTR]: true}"
               class="quick-navigation__refs"
+              @keydown.down.exact.prevent="focusNext"
+              @keydown.up.exact.prevent="focusPrev"
+              @keydown.enter.exact="handleKeyEnter"
+              @click.self="closeQuickNavigationModal"
```
Comment (dobromir-hristov) [inline src/components/Navigator/QuickNavigationModal.vue:65]: Not sure this click handler does much 🤔 

The modal already handles closing, when you click outside.

**Review by dobromir-hristov — COMMENTED** [review]
I am getting weird behaviour, where the focus skips the items on the left, but on focus-back, it focuses them.
Also if I click `down-arrow` I see the page changing, but not the items. 

On Safari I also cant focus on the `X` button to clear the input.

File: src/components/Navigator/QuickNavigationModal.vue (hunk 1 above)
Comment (marinaaisa) [reply src/components/Navigator/QuickNavigationModal.vue]: Fixed at https://github.com/apple/swift-docc-render/pull/736/commits/8d8c17e136086ff633c105f2f4aa570e54361551

File: src/components/Navigator/QuickNavigationModal.vue (hunk 3 above)
Comment (marinaaisa) [reply src/components/Navigator/QuickNavigationModal.vue]: https://github.com/apple/swift-docc-render/pull/736/commits/8d8c17e136086ff633c105f2f4aa570e54361551

File: src/components/Navigator/QuickNavigationModal.vue (hunk 2 above)
Comment (marinaaisa) [reply src/components/Navigator/QuickNavigationModal.vue]: Not using it anymore https://github.com/apple/swift-docc-render/pull/736/commits/8d8c17e136086ff633c105f2f4aa570e54361551

**Comment by marinaaisa** [comment]
> I am getting weird behaviour, where the focus skips the items on the left, but on focus-back, it focuses them. Also if I click `down-arrow` I see the page changing, but not the items.
> 
> On Safari I also cant focus on the `X` button to clear the input.

Hey @dobromir-hristov thanks for your feedback, there was a bug going on.
Can you test it again? It should be fine now.

**Comment by marinaaisa** [comment]
@swift-ci test

**Comment by marinaaisa** [comment]
@swift-ci test

**Review by mportiz08 — APPROVED** [review]
Seems to be working well from my testing :)

**Comment by marinaaisa** [comment]
@swift-ci test

**Pull request merged**